lib/formulas/resource-router-mapping.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+var recast = require('recast');
+var builders = recast.types.builders;
+var types = recast.types.namedTypes;
+
+module.exports = function transformResourceRouterMapping(source) {
+  var ast = recast.parse(source);
+
+  recast.visit(ast, {
+    visitCallExpression: function (path) {
+      var node = path.node;
+
+      if (isRouter(node)) {
+        var routes = node.arguments[0].body.body;
+        traverseExpressions(routes);
+        return false;
+      } else {
+        this.traverse(path);
+      }
+    }
+  });
+
+  return recast.print(ast, {tabWidth: 2, quote: 'single'}).code;
+};
+
+function traverseExpressions(routes) {
+  routes.forEach(function (route) {
+    if (isRoute(route)) {
+      transformResourceToRoute(route);
+      var lastArgument = route.expression.arguments[route.expression.arguments.length-1];
+      if (lastArgument.type === 'FunctionExpression') {
+        traverseExpressions(lastArgument.body.body);
+      }
+    }
+  });
+}
+
+function ensureOptionsObject(route) {
+  var args = route.expression.arguments;
+  var secondArg = args[1];
+
+  if (secondArg && types.ObjectExpression.check(secondArg)) {
+    return secondArg;
+  }
+
+  var optionsObject = builders.objectExpression([]);
+  args.splice(1, 0, optionsObject);
+
+  return optionsObject;
+}
+
+function transformResourceToRoute(route) {
+  var expression = route.expression;
+  var callee = expression.callee;
+  var optionsObject = ensureOptionsObject(route);
+  var resetNamespaceOption = builders.property(
+    'init',
+    builders.identifier('resetNamespace'),
+    builders.literal(true)
+  );
+
+  callee.property.name = 'route';
+  optionsObject.properties.push(resetNamespaceOption);
+}
+
+function isRoute(route) {
+  return route.type === 'ExpressionStatement' &&
```

Comment:
@marcioj can you use recast's API to check the type ?

You can do something like

```
var types            = recast.types.namedTypes;

type. ExpressionStatement.check(route)
type.CallExpression.check(route.expression.callee)
```